packages/terra-dev-site/src/webpack/plugin/SitePlugin.js

```diff
@@ -341,6 +359,72 @@ class SitePlugin {
       oneTimeSetupComplete = false;
       siteRegistry = {};
     });
+
+    // Evaluate all URLS found in the site for correctness.
+    compiler.hooks.done.tap('terra-dev-site-plugin', () => {
+      // Convert generated links for all sites into a single list.
+      const validRelativeUrls = Object.values(generatedBootstrapData).reduce((acc, urls) => acc.concat(urls), []);
+      // add the favicon since it's valid too.
+      validRelativeUrls.push(`${publicPath.slice(0, -1)}/favicon.ico`);
+
+      // Log formatting
+      const invalidUrlLog = (url) => `${chalk.blue(url.url)} in:\n ${chalk.yellow(url.file.history[0].replace(url.file.cwd, ''))} at line ${url.node.position.start.line}`;
+
+      // Create a promise array to allow all logs to happen at the same time.
+      const resultPromises = [];
+      const queueLength = urlQueue.length;
+      for (let index = 0; index < queueLength; index += 1) {
+        const url = urlQueue.shift();
+
+        if (url.url.startsWith('/')) {
+          // for relative urls check against the know list of valid relative urls.
+          // Remove query params or hash links from the end of the url.
+          const urlWithoutHash = url.url.replace(/[#?].*$/, '');
+          if (!validRelativeUrls.includes(urlWithoutHash)) {
+            resultPromises.push(Promise.resolve(`${chalk.red('[Broken Link]')} ${invalidUrlLog(url)}`));
```

Comment:
this is returned with a promise to enable http or https checking in the future that must be asynchronous. this has the added benefit of not slowing down webpack.